ra/ra.go

```diff
@@ -361,6 +364,33 @@ func (ra *RegistrationAuthorityImpl) NewAuthorization(ctx context.Context, reque
 		}
 	}
 
+	if ra.reuseValidAuthz {
+		auths, err := ra.SA.GetValidAuthorizations(ctx, regID, []string{identifier.Value}, ra.clk.Now())
+		if err != nil {
+			outErr := core.InternalServerError("unable to get existing validations")
+			ra.log.Warning(string(outErr))
+		}
+
+		if existingAuthz, ok := auths[identifier.Value]; ok {
+			// Use the valid existing authorization's ID to find a fully populated version
+			// The results from `GetValidAuthorizations` are most notably missing
+			// `Challenge` values that the client expects in the result.
+			populatedAuthz, err := ra.SA.GetAuthorization(ctx, existingAuthz.ID)
+			if err != nil {
+				outErr := core.InternalServerError("unable to get existing authorization")
+				ra.log.Warning(fmt.Sprintf("%s: %s", string(outErr), existingAuthz.ID))
+			}
+
+			// The existing authorization must not expire within the next 24 hours for
+			// it to be OK for reuse
+			reuseCutOff := ra.clk.Now().AddDate(0, 1, 0)
```

Comment:
While this is perfectly fine `.Add(time.Hour * 24)` seems a little easier to read.